shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -1822,6 +1837,53 @@ private AccessibilityEvent obtainAccessibilityEvent(int virtualViewId, int event
     return event;
   }
 
+  /**
+   * Reads the layoutInDisplayCutoutMode value from the window attribute and returns whether a left
+   * cutout insect is required.
+   *
+   * <p>The layoutInDisplayCutoutMode is added after API level 28.
+   */
+  @TargetApi(28)
+  @RequiresApi(28)
+  private boolean doesLayoutInDisplayCutoutModeRequireLeftInsect() {
+    Context context = rootAccessibilityView.getContext();
+    Activity activity = getActivity(context);
+    if (activity != null) {
+      if (activity.getWindow() == null) {
+        // The activity is not visible, it does not matter whether to apply left insect
+        // or not.
+        return false;
+      }
+      int layoutInDisplayCutoutMode =
+          activity.getWindow().getAttributes().layoutInDisplayCutoutMode;
+      return layoutInDisplayCutoutMode
+              == WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_NEVER
+          || layoutInDisplayCutoutMode
+              == WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_DEFAULT;
+    }
+    return true;
+  }
+
+  /**
+   * Retrieves the {@link Activity} from a given {@link Context}.
+   *
+   * <p>This method will recursively traverse up the context chain if it is a {@link ContextWrapper}
+   * until it finds the first instance of the base context that is an {@link Activity}.
+   */
+  private static Activity getActivity(Context context) {
```

Comment:
It feels weird to put it here, but I don't want to expose a public api on embedding/engine/FlutterView.java.

Reply:
These packages are structured in a weird way. e.g. `AccessibilityBridge` should be in the same package as `FlutterView`.  @jason-simmons do you have any context into why this is the case? Also, do you have any concerns if these are merged?

Reply:
There's a `io/flutter/util` folder. Do you mind if we create a `ViewUtil` class and move this static method there, and replace the one from `FlutterView` as well?

Reply:
done